### PR TITLE
Fix some warnings found by valgrind

### PR DESCRIPTION
**Pull request by DolceTriade**

### CODE REVIEW

File: src/engine/qcommon/net_chan.cpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ Send one fragment of the current message
 void Netchan_TransmitNextFragment( netchan_t *chan )
 {
 	msg_t send;
-	byte  send_buf[ MAX_PACKETLEN ];
+	byte  send_buf[ MAX_PACKETLEN ] = {0};
```
Comment (slipher) [inline src/engine/qcommon/net_chan.cpp:120]: Why is this needed? The code is designed to memset the buffer - see Huff_putBit.

File: src/common/Serialize.h [hunk 2]
```diff
@@ -74,6 +74,10 @@ namespace Util {
 	public:
 		void WriteData(const void* p, size_t len)
 		{
+			if (data.capacity() < data.size() + len)
```
[the remark this answers is not in the document]
Comment (DolceTriade) [reply src/common/Serialize.h:77]: fwiw, this warning prevents compilation using -Werror, so we either need to disable this warning or fix the code.

How does it cause O(N^2) for repeated small writes? If you push_back when you go above the capacity(), doesn't that cause a reallocation anyways?

File: src/engine/qcommon/net_chan.cpp (hunk 1 above)
Comment (DolceTriade) [reply src/engine/qcommon/net_chan.cpp:120]: Valgrind still complains :shrug:

**Comment by DolceTriade** [comment]
These aren't critical fixes. Let's just close this for now.
